lib/modules/manager/conan/artifacts.spec.ts

```diff
@@ -0,0 +1,243 @@
+  localDir: join('/tmp/github/some/repo'),
+  cacheDir: join('/tmp/cache'),
+  containerbaseDir: join('/tmp/cache/containerbase'),
+  dockerSidecarImage: 'ghcr.io/containerbase/sidecar',
+};
+
+describe('modules/manager/conan/artifacts', () => {
+  beforeEach(() => {
+    env.getChildProcessEnv.mockReturnValue(envMock.basic);
+    GlobalConfig.set(adminConfig);
+    docker.resetPrefetchedImages();
+    hostRules.getAll.mockReturnValue([]);
+  });
+
+  afterEach(() => {
+    GlobalConfig.reset();
+  });
+
+  it('returns null if conan.lock maintenance is turned off', async () => {
+    const updatedDeps = [
+      {
+        depName: 'dep',
+      },
+    ];
+
+    expect(
+      await conan.updateArtifacts({
+        packageFileName: 'conanfile.py',
+        updatedDeps,
+        newPackageFileContent: '',
+        config,
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if there are no dependencies to update', async () => {
+    expect(
+      await conan.updateArtifacts({
+        packageFileName: 'conanfile.py',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config: { ...config, updateType: 'lockFileMaintenance' },
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if conan.lock was not found', async () => {
+    const updatedDeps = [
+      {
+        depName: 'dep',
+      },
+    ];
+
+    fs.findLocalSiblingOrParent.mockResolvedValueOnce(null);
+
+    expect(
+      await conan.updateArtifacts({
+        packageFileName: 'conanfile.py',
+        updatedDeps,
+        newPackageFileContent: '',
+        config: { ...config, updateType: 'lockFileMaintenance' },
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if conan.lock read operation failed', async () => {
+    const updatedDeps = [
+      {
+        depName: 'dep',
+      },
+    ];
+
+    fs.findLocalSiblingOrParent.mockResolvedValueOnce('conan.lock');
+    fs.readLocalFile.mockResolvedValueOnce(null);
+
+    expect(
+      await conan.updateArtifacts({
+        packageFileName: 'conanfile.py',
+        updatedDeps,
+        newPackageFileContent: '',
+        config: { ...config, updateType: 'lockFileMaintenance' },
+      }),
+    ).toBeNull();
+  });
+
+  it('returns null if read operation failed for new conan.lock', async () => {
+    const updatedDeps = [
+      {
+        depName: 'dep',
+      },
+    ];
+
+    fs.statLocalFile.mockResolvedValueOnce({ name: 'conan.lock' } as any);
+    fs.findLocalSiblingOrParent.mockResolvedValueOnce('conan.lock');
+    fs.readLocalFile.mockResolvedValueOnce('Original conan.lock');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce(null);
+
+    expect(
+      await conan.updateArtifacts({
+        packageFileName: 'conanfile.py',
+        updatedDeps,
+        newPackageFileContent: '',
+        config: { ...config, updateType: 'lockFileMaintenance' },
+      }),
+    ).toBeNull();
+    expect(execSnapshots).toMatchSnapshot();
```

Comment:
nope, checkout other managers for `toMatchObject` samples for this 

https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md#unit-testing